pathio.go

```diff
@@ -117,12 +121,22 @@ func parseS3Path(path string) (string, string, error) {
 	return bucketName, key, nil
 }
 
-// getRegion converts a region name into an aws.Region object
-func region(regionString string) (aws.Region, error) {
-	for name, region := range aws.Regions {
-		if strings.ToLower(name) == strings.ToLower(regionString) {
-			return region, nil
-		}
+// getRegionForBucket looks up the region name for the given bucket
+func getRegionForBucket(name string) (string, error) {
+	// Any region will work for the region lookup, but the request MUST use
+	// PathStyle
+	config := aws.NewConfig().WithRegion("us-west-1").WithS3ForcePathStyle(true)
+	client := s3.New(config)
+	params := &s3.GetBucketLocationInput{
+		Bucket: aws.String(name),
```

Comment:
like thrift, amazon uses string pointers, so the aws.String() is just a convenience function that allows us to get a pointer from a string literal since golang doesn't let us do it inline.